server/leader.go

```diff
@@ -129,8 +139,12 @@ func (s *Server) marshalLeader() string {
 	return string(data)
 }
 
+func (s *Server) isEtcdLeader() bool {
+	return s.etcd.Server.ID() == s.etcd.Server.Leader()
+}
```

Comment:
I don't know the reason but etcd indicates that `etcdServer.Leader()` is supposed to be only used for testing. See https://github.com/coreos/etcd/blob/master/etcdserver/server.go#L856
Could cause any problem here?